src/visibility.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+import {BasePlugin, Utils} from 'playkit-js';
+import './style.css';
+import {DismissibleFloatingButtonComponent} from './components/dismissible/dismissible';
+
+const FLOATING_ACTIVE_CLASS: string = 'playkit-floating-active';
+const FLOATING_CONTAINER_CLASS: string = 'playkit-floating-container';
+const FLOATING_POSTER_CLASS: string = 'playkit-floating-poster';
+
+/**
+ * Visibility class.
+ * @classdesc
+ */
+class Visibility extends BasePlugin {
+  appTargetContainer: HTMLElement | null;
+  floatingContainer: HTMLElement;
+  floatingPoster: HTMLElement;
+  observer: window.IntersectionObserver;
+  everStartedPlaying: boolean = false;
+  dismissed: boolean = false;
+
+  /**
+   * The default configuration of the plugin.
+   * @type {Object}
+   * @static
+   */
+  static defaultConfig: Object = {
+    threshold: 0.5
+  };
+
+  getUIComponents() {
+    return [
+      {
+        label: 'dismissibleFloatingButtonComponent',
+        presets: ['Playback', 'Live'],
+        container: 'TopBarRightControls',
+        get: DismissibleFloatingButtonComponent,
+        props: {
+          onClose: () => {
+            this._dismissed();
+          }
+        }
+      }
+    ];
+  }
+
+  /**
+   * @static
+   * @public
+   * @returns {boolean} - Whether the plugin is valid.
+   */
+  static isValid(): boolean {
+    return true;
+  }
+
+  /**
+   * @constructor
+   * @param {string} name - The plugin name.
+   * @param {Player} player - The player instance.
+   * @param {Object} config - The plugin config.
+   */
+  constructor(name: string, player: Player, config: Object) {
+    super(name, player, config);
+    this.reset();
+    if (this.config.floating) {
+      this._initFloating();
+    }
+    const options = {
+      threshold: this.config.threshold
+    };
+
+    this.observer = new window.IntersectionObserver(this._handleVisibilityChange.bind(this), options);
```

Comment:
I think the `IntersectionObserver` should be inited on `loadMedia` and clear in `reset`, it shouldn't work until `loadMedia` called anyway

Reply:
clear in `destroy` not in `reset`. you won't reset it in change media

Reply:
The IntersectionOberserver should be initiated in the constructor as other functionalities like AutoPlayPause might be needed before loadMedia. In addition, when switching videos in a playlist we don't we don't want to freeze the floating mechanism.